packages/governance/src/binaryBallotCounter.js

```diff
@@ -23,60 +27,71 @@ const makeQuorumCounter = quorumThreshold => {
-  question,
-  positions,
-  threshold,
-  tieOutcome = undefined,
-  closingRule,
-  instance,
-) => {
+const validateBinaryBallotSpec = ballotSpec => {
+  looksLikeBallotSpec(ballotSpec);
+
+  const positions = ballotSpec.positions;
   assert(
     positions.length === 2,
     X`Binary ballots must have exactly two positions. had ${positions.length}: ${positions}`,
   );
-  assert.typeof(question, 'string');
-  assert.typeof(positions[0], 'string');
-  assert.typeof(positions[1], 'string');
-  if (tieOutcome) {
-    assert(
-      positions.includes(tieOutcome),
-      X`The default outcome on a tie must be one of the positions, not ${tieOutcome}`,
-    );
-  }
-
-  const template = buildBallot(
-    ChoiceMethod.CHOOSE_N,
-    question,
-    positions,
-    1,
-    instance,
-  );
-  const ballotDetails = template.getDetails();
 
   assert(
-    ballotDetails.method === ChoiceMethod.CHOOSE_N,
-    X`Binary ballot counter only works with CHOOSE_N`,
+    ballotSpec.maxChoices === 1,
+    X`Can only choose 1 item on a binary ballot`,
+  );
+  assert(
+    ballotSpec.method === ChoiceMethod.APPROVAL,
+    X`${ballotSpec.method} must be APPROVAL`,
   );
+  // We don't check the quorumRule or quorumThreshold here. The quorumThreshold
+  // is provided by the Registrar that creates this ballotCounter, since only it
+  // can translate the quorumRule to a required number of votes.
+};
+
+// Notice that BinaryBallotCounter is designed to run as a Zoe contract. The
+// business part of the contract is extracted here so it can be tested
+// independently. The standard Zoe start function is at the bottom of this file.
+
+/** @type {BuildBallotCounter} */
+const makeBinaryBallotCounter = (ballotSpec, threshold, instance) => {
+  validateBinaryBallotSpec(ballotSpec);
+
+  const ballotTemplate = buildApprovalBallot(ballotSpec, instance);
+  const details = ballotTemplate.getDetails();
+  const { handle } = details;
+
   let isOpen = true;
+  const positions = ballotSpec.positions;
   const outcomePromise = makePromiseKit();
   const tallyPromise = makePromiseKit();
-  const allBallots = makeStore('seat');
-
-  const recordBallot = (seat, filledBallotP, shares = 1n) => {
+  // The Registrar is responsible for creating a unique seat for each voter.
+  // This ballotCounter allows voters to re-vote, and replaces their previous
+  // choice with the new selection.
+
+  /**
+   * @typedef {Object} RecordedBallot
+   * @property {CompleteApprovalBallot} ballot
+   * @property {bigint} shares
+   */
+  /** @type {Store<Handle<'Voter'>,RecordedBallot> } */
+  const allBallots = makeStore('handle');
+
+  /** @type {SubmitVote} */
+  const submitVote = (voterHandle, filledBallotP, shares = 1n) => {
     return E.when(filledBallotP, filledBallot => {
       assert(
-        filledBallot.question === question,
-        X`Ballot not for this question ${filledBallot.question} should have been ${question}`,
+        filledBallot.handle === handle,
```

Comment:
I would prefer for variables named `handle` to instead be `somethingHandle`. Otherwise `handle` is too much like "thing".